playground/backend/cmd/server/controller.go

```diff
@@ -271,43 +285,45 @@ func cleanUp(pipelineId uuid.UUID, lc *fs_tool.LifeCycle) {
 // processError processes error received during processing code via setting a corresponding status and output to cache
 func processError(ctx context.Context, err error, data []byte, pipelineId uuid.UUID, cacheService cache.Cache, status pb.Status) {
 	switch status {
-	case pb.Status_STATUS_ERROR:
-		logger.Errorf("%s: Run: err: %s, output: %s\n", pipelineId, err.Error(), data)
-
-		// set to cache pipelineId: cache.SubKey_RunOutput: err.Error()
-		setToCache(ctx, cacheService, pipelineId, cache.RunOutput, "error: "+err.Error()+", output: "+string(data))
+	case pb.Status_STATUS_VALIDATION_ERROR:
+		logger.Errorf("%s: Validate: %s\n", pipelineId, err.Error())
 
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_ERROR
-		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_ERROR)
+		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_VALIDATION_ERROR)
 	case pb.Status_STATUS_COMPILE_ERROR:
 		logger.Errorf("%s: Compile: err: %s, output: %s\n", pipelineId, err.Error(), data)
 
-		// set to cache pipelineId: cache.SubKey_CompileOutput: err.Error()
 		setToCache(ctx, cacheService, pipelineId, cache.CompileOutput, "error: "+err.Error()+", output: "+string(data))
 
-		// set to cache pipelineId: cache.SubKey_Status: pb.Status_STATUS_ERROR
 		setToCache(ctx, cacheService, pipelineId, cache.Status, pb.Status_STATUS_COMPILE_ERROR)
+	case pb.Status_STATUS_RUN_ERROR:
+		logger.Errorf("%s: Run: err: %s, output: %s\n", pipelineId, err.Error(), data)
+
+		setToCache(ctx, cacheService, pipelineId, cache.RunError, "error: "+err.Error()+", output: "+string(data))
+		setToCache(ctx, cacheService, pipelineId, cache.RunOutput, "")
```

Comment:
Should we really set a `cache.RunOutput`? Can't we skip this step?

Reply:
Changed.